src/Spice86.Core/Emulator/InterruptHandlers/Input/Keyboard/BiosKeyboardInt9Handler.cs

```diff
@@ -6,35 +6,96 @@
 using Spice86.Core.Emulator.Devices.ExternalInput;
 using Spice86.Core.Emulator.Devices.Input.Keyboard;
 using Spice86.Core.Emulator.Function;
+using Spice86.Core.Emulator.InterruptHandlers.Common.MemoryWriter;
 using Spice86.Core.Emulator.Memory;
+using Spice86.Shared.Emulator.Memory;
 using Spice86.Shared.Interfaces;
 
 /// <summary>
-/// Crude implementation of BIOS keyboard buffer handler (hardware interrupt 0x9, IRQ1)
+/// Implementation of BIOS keyboard buffer handler (hardware interrupt 0x9, IRQ1)
 /// </summary>
 public class BiosKeyboardInt9Handler : InterruptHandler {
-    private readonly Keyboard _keyboard;
+    private readonly AvaloniaKeyConverter _scanCodeConverter = new();
```

Comment:
Why is bios dealing with avalonia? shouldnt this be done at the device level?

Reply:
This is temporary, the ps2 keyboard is supposed to have several scan code sets, and it's INT9H handler that is supposed to convert scan code into ASCII.

But this alone would bloat the PR by thousands of lines.